### ISSUE TITLE
Emulate CREATE SCHEMA with CREATE USER .. NO AUTHENTICATION in Oracle 18c

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15307]
Oracle 18c introduced formal support for "schema only" accounts, i.e. effectively supporting their own version of `CREATE SCHEMA`:

- https://oracle-base.com/articles/18c/schema-only-accounts-18c
- https://docs.oracle.com/en/database/oracle/oracle-database/18/dbseg/configuring-authentication.html#GUID-22328DCF-B137-499B-ACC9-49D6C72B6A57

The syntax is:

```sql
CREATE USER my_schema NO AUTHENTICATION;
```

In older versions of Oracle, we'll simply omit the clause, not supplying password, etc.

Related commands include:

```sql
DROP USER my_schema;
```

There's no `ALTER USER` command:

```sql
ALTER USER my_schema RENAME TO my_other_schema;
```

Result:

> SQL Error [3001] [0A000]: ORA-03001: unimplemented feature

**Comment by lukaseder** [comment]
There are some Oracle specific details caused by the fact that a `NO AUTHENTICATION` user is still a user with grants (or lack of grants). For example, this doesn't work:

```sql
create user "s1" no authentication;
create user "s2" no authentication;
create table "s1"."t1" ("c1" number(10) not null, primary key ("c1"));
create table "s2"."t2" ("c2" number(10) not null);
alter table "s2"."t2" add foreign key ("c2") references "s1"."t1" ("c1");
```

In order to add the `FOREIGN KEY`, the user needs the following `GRANT`:

```sql
grant references on "s1"."t1" to "s2";
```

Of course, users could be more blunt and just grant `any privileges`, hoping the user can never actually connect.

From a schema only perspective, this all doesn't make sense, but I'm not sure if jOOQ can really automate anything here without creating weird security holes in an Oracle database.

**Comment by lukaseder** [comment]
For the schema to be used meaningfully, we have to `QUOTA UNLIMITED ON USERS`. This is again a bit weird to default to, as users might not want things this way, and we're currently not offering any such storage options.

I've created a feature request for `TABLESPACE` clause support of various statements:

- https://github.com/jOOQ/jOOQ/issues/15308

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.19.0

**Comment by lukaseder** [comment]
There's a follow-up task to manage privileges among schemas created this way:

- https://github.com/jOOQ/jOOQ/issues/17342

I'm not sure yet how / if this follow-up task should be implemented.
